src/main/kotlin/com/openlattice/entitysets/controllers/EntitySetsController.kt

```diff
@@ -548,58 +561,73 @@ constructor(
-            entitySetManager.deleteEntitySet(it)
-            securableObjectTypes.deleteSecurableObjectType(AclKey(it))
+        val authorizedOrganizationIds = authorizations
+                .accessChecksForPrincipals(
+                        organizationIds.map { AccessCheck(AclKey(it), READ_PERMISSION) }.toSet(),
+                        Principals.getCurrentPrincipals()
+                )
+                .filter { it.permissions.contains(Permission.READ) }
+                .map { it.aclKey[0] }
+                .collect(Collectors.toSet())
+
+        if (authorizedOrganizationIds.size < organizationIds.size) {
+            throw ForbiddenException(
+                    "Can't create entity sets, missing read permissions on organizations with ids " +
+                            "${organizationIds.subtract(authorizedOrganizationIds)}"
+            )
         }
 
-        aresManager.removeAuditRecordEntitySetConfiguration(aclKey)
-    }
+        // if it's a linking entity set, check link on linked entity sets
+        val linkedEntitySetIds = entitySets.filter { it.isLinking }.flatMap { it.linkedEntitySets }.toSet()
 
-    override fun getAuthorizationManager(): AuthorizationManager {
-        return authorizations
+        val authorizedLinkedEntitySetIds = authorizations
+                .accessChecksForPrincipals(
+                        linkedEntitySetIds.map { AccessCheck(AclKey(it), EnumSet.of(Permission.LINK)) }.toSet(),
+                        Principals.getCurrentPrincipals()
+                )
+                .filter { it.permissions.contains(Permission.LINK) }
+                .map { it.aclKey[0] }
+                .collect(Collectors.toSet())
+
+        if (authorizedLinkedEntitySetIds.size < linkedEntitySetIds.size) {
+            throw ForbiddenException(
+                    "Can't create linking entity entity sets, missing link permissions on linked entity sets with " +
+                            "ids ${linkedEntitySetIds.subtract(authorizedLinkedEntitySetIds)}"
+            )
+        }
     }
 
-    private fun ensureEntitySetCanBeDeleted(entitySet: EntitySet) {
-        val entitySetId = entitySet.id
+    private fun checkPermissionsForDelete(entitySetId: UUID): EntitySet {
+        ensureOwnerAccess(AclKey(entitySetId))
 
-        ensureObjectCanBeDeleted(entitySetId)
+        val entitySet = entitySetManager.getEntitySet(entitySetId)!!
+        val entityType = edmManager.getEntityType(entitySet.entityTypeId)
 
-        if (entitySet.flags.contains(EntitySetFlag.AUDIT)) {
-            throw ForbiddenException("You cannot delete entity set $entitySetId because it is an audit entity set.")
+        val authorizedPropertyTypes = authzHelper.getAuthorizedPropertyTypes(entitySetId, EnumSet.of(Permission.OWNER))
+        if (!authorizedPropertyTypes.keys.containsAll(entityType.properties)) {
+            throw ForbiddenException("You shall not pass!")
```

Comment:
I realize this was just moved around during a refactor, but maybe this error message can be a little more clear? :)